File: solutions/lua/darts/2/darts.lua
local Darts = {}

local function radius(x, y)
     return math.sqrt(x ^ 2 +  y ^ 2)
end

function Darts.score(x, y)
    local r = radius(x, y)
    if r > 10 then
        return 0
    elseif r > 5 then
        return 1
    elseif r > 1 then
        return 5
    else
        return 10
    end
end

return Darts

File: solutions/java/darts/1/src/main/java/Darts.java
class Darts {

    private static final int OUTER_CIRCLE_RADIUS = 10;
    private static final int MIDDLE_CIRCLE_RADIUS = 5;
    private static final int INNER_CIRCLE_RADIUS = 1;

    private double radius;

    Darts(double x, double y) {
        radius = Math.sqrt(x * x + y * y);
    }

    int score() {
        if (radius <= INNER_CIRCLE_RADIUS)
            return 10;
        if (radius <= MIDDLE_CIRCLE_RADIUS)
            return 5;
        if (radius <= OUTER_CIRCLE_RADIUS)
            return 1;

        return 0;
    }
}
